### DISCUSSION TITLE
Ddata Race

### DISCUSSION THREAD

**Discussion by manson** [kamalshkeir/kago#16] [General]
Hi!
I'm doing some multithreading usage of all this. It looks like that Logger is not thread safe. I have Data Race:
```
==================
WARNING: DATA RACE
Write at 0x0000019ac190 by goroutine 8:
  github.com/kamalshkeir/kago/core/utils/logger.Info()
      /home/manson/go/pkg/mod/github.com/kamalshkeir/kago@v1.2.7/core/utils/logger/logger.go:124 +0x384
  authors.city/pss/src/lib/sse.(*Broker).Start.func1()
      /home/manson/work/authors.city/payments/showcase_server/src/lib/sse/broker.go:99 +0x384 // here in my code I do logger.Info("Added new client")

Previous read at 0x0000019ac190 by goroutine 7:
  github.com/kamalshkeir/kago/core/utils/logger.init.0.func1()
      /home/manson/go/pkg/mod/github.com/kamalshkeir/kago@v1.2.7/core/utils/logger/logger.go:25 +0x30
  github.com/kamalshkeir/kago/core/utils/eventbus.Subscribe[...].func1()
      /home/manson/go/pkg/mod/github.com/kamalshkeir/kago@v1.2.7/core/utils/eventbus/eventbus.go:51 +0x52
```
This is all around using the StreamLogs inside the Logger. I looked the code - it really is not safe. It needs to be wrapped by a mutex or something like that. Without this app hangs at some point

**Comment by kamalshkeir** [comment]
@manson , it has been longtime since kago, i will for sure make it as deprecated, replaced with korm or a new one, also some dependency like x/net should be updated , there is a vulnerability found in it, i did it for kmux but not yet for kago , can you try do what you are doing using kmux and korm ?

**Reply by manson** [reply]
Oh, I see. I took it from examples and it automatically attached from kago module... Will check it out
